### PR TITLE
chore: change postal and postalExt to string

### LINKED ISSUE #1094: Geocoding postal code type
### Description

Jim Barry brought to my attention that the postal code parameter should be wrapped in quotes as it is an alphanumeric value. Issues arise when using a postal code starting with zero as demonstrated in these CodePen examples ([ex1](https://codepen.io/cyatteau/pen/OJoYRNr), [ex2](https://codepen.io/cyatteau/pen/BaOeRLj)). 

Several places state or use a _number_ instead of a _string_, which need to be addressed:
- [ ] The [API Reference](https://developers.arcgis.com/arcgis-rest-js/api-reference/arcgis-rest-geocoding/IGeocodeOptions#postal) states that the postal code value should be a _number_.
- [ ] The example given [here](https://developers.arcgis.com/arcgis-rest-js/api-reference/arcgis-rest-geocoding/geocode).
- [ ] The tutorial [here](https://developers.arcgis.com/arcgis-rest-js/geocode-and-search/search-for-an-address/).

### Link to page

https://developers.arcgis.com/arcgis-rest-js/api-reference/arcgis-rest-geocoding/IGeocodeOptions#postal

### PR DESCRIPTION
**Pull request by BrunoCaimar**
Fix #1094

### CODE REVIEW

**Comment by BrunoCaimar** [comment]
Changed to accept string or number. Also, added tests to cover the change.

File: packages/arcgis-rest-geocoding/test/bulk.test.ts [hunk 1]
```diff
@@ -159,6 +159,51 @@ describe("geocode", () => {
...
+        expect(url).toEqual(
+          "https://customer.gov/arcgis/rest/services/CompositeGeocoder/GeocodeServer/geocodeAddresses"
+        );
+        expect(options.method).toBe("POST");
+        expect(options.body).toContain("f=json");
+        expect(options.body).toContain("outSR=4326");
+        expect(options.body).toContain("forStorage=true");
+        expect(options.body).toContain(
+          `addresses=${encodeURIComponent(
+            '{"records":[{"attributes":{"OBJECTID":1,"address":"380 New York St. Redlands","postal":92373}},{"attributes":{"OBJECTID":2,"address":"1205 Williston Rd","postal":"05403"}}]}'
```
Comment (gavinr-maps) [inline packages/arcgis-rest-geocoding/test/bulk.test.ts:199]: @patrickarlt for cases of batch geocoding, is it ok that we could be sending the `postal` as both a string and number? Or should we be making that consistent within ArcGIS REST JS before it makes the rest call?

**Review by gavinr-maps — COMMENTED** [review]
Overall looks good. Thanks @BrunoCaimar!

I had one question that I wanted to ask @patrickarlt what he thinks:

File: packages/arcgis-rest-geocoding/test/bulk.test.ts (hunk 1 above)
Comment (patrickarlt) [reply packages/arcgis-rest-geocoding/test/bulk.test.ts:199]: I don't think so, the server should be able to sort it out. Lets merge this.

**Review by gavinr-maps — APPROVED** [review]

**Pull request merged**